test/crit/main.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/checkpoint-restore/go-criu/v5/crit"
+)
+
+func main() {
+	imgs := getImgs()
+	for _, img := range imgs {
+		log.Println("=== ", img)
+		testImg := fmt.Sprintf("%s.test.img", filepath.Base(img))
+		c := crit.New(img, testImg, "", false, false)
+		// Decode the binary image file
+		decodedImg, err := c.Decode()
+		if err != nil {
+			log.Fatal("[DECODE]: ", err)
+			continue
+		}
+		// Encode it into test binary image file
+		err = c.Encode(decodedImg)
+		if err != nil {
+			log.Fatal("[ENCODE]: ", err)
+			continue
+		}
+		// Open and compare original and test files
+		imgBytes, err := os.ReadFile(img)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+		testImgBytes, err := os.ReadFile(testImg)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+		if !bytes.Equal(imgBytes, testImgBytes) {
+			log.Fatal("[RECODE]: files do not match")
+		}
+		log.Println("=== PASS")
+	}
+}
+
+func getImgs() []string {
+	var skipImgs = []string{
+		"pages-",
+		"iptables-",
+		"ip6tables-",
+		"nftables-",
+		"route-",
+		"route6-",
+		"ifaddr-",
+		"tmpfs-",
+		"netns-ct-",
+		"netns-exp-",
+		"rule-",
+	}
```

Comment:
Should we include all raw images?

```bash
$ grep FD_ENTRY_F criu/image-desc.c 
#define FD_ENTRY_F(_name, _fmt, _f)     \
	FD_ENTRY_F(PIPES_DATA,	"pipes-data", O_NOBUF), /* splices data */
	FD_ENTRY_F(FIFO_DATA,	"fifo-data", O_NOBUF), /* the same */
	FD_ENTRY_F(SK_QUEUES,	"sk-queues", O_NOBUF), /* lseeks the image */
	FD_ENTRY_F(IPCNS_SHM,	"ipcns-shm-%u", O_NOBUF), /* writes segments of data */
	FD_ENTRY_F(GHOST_FILE,	"ghost-file-%x", O_NOBUF),
	FD_ENTRY_F(MEMFD_INODE,	"memfd", O_NOBUF),
	FD_ENTRY_F(IFADDR,	"ifaddr-%u", O_NOBUF),
	FD_ENTRY_F(ROUTE,	"route-%u", O_NOBUF),
	FD_ENTRY_F(ROUTE6,	"route6-%u", O_NOBUF),
	FD_ENTRY_F(RULE,	"rule-%u", O_NOBUF),
	FD_ENTRY_F(IPTABLES,	"iptables-%u", O_NOBUF),
	FD_ENTRY_F(IP6TABLES,	"ip6tables-%u", O_NOBUF),
	FD_ENTRY_F(NFTABLES,	"nftables-%u", O_NOBUF),
	FD_ENTRY_F(TMPFS_IMG,	"tmpfs-%u.tar.gz", O_NOBUF),
	FD_ENTRY_F(TMPFS_DEV,	"tmpfs-dev-%u.tar.gz", O_NOBUF),
	FD_ENTRY_F(AUTOFS,	"autofs-%u", O_NOBUF),
	FD_ENTRY_F(TTY_DATA,	"tty-data", O_NOBUF),
	FD_ENTRY_F(PAGES,	"pages-%u", O_NOBUF),
	FD_ENTRY_F(PAGES_OLD,	"pages-%d", O_NOBUF),
	FD_ENTRY_F(SHM_PAGES_OLD, "pages-shmem-%ld", O_NOBUF),
	FD_ENTRY_F(BPFMAP_FILE,	"bpfmap-file", O_NOBUF),
	FD_ENTRY_F(BPFMAP_DATA,	"bpfmap-data", O_NOBUF),
```

Reply:
`Decode()` and `Encode()` support some of these images, like ghost files, pipes data, etc. So I think we can retain those. I'll test which ones are not supported and add them in 👍